### ISSUE TITLE
Program icons arranged to stay visible are being hidden

### ISSUE THREAD

**Issue by Markospoko** [dwarvesf/hidden#181]
### Describe the bug

Icons that are arranged to stay displayed (between | and >) are displayed fine but after restarting Mac it's faulty and they become hidden in older High Sierra.

### Steps to reproduce

1. Arrange icons that you want to see between | and >, click shrink icon >, it's all fine
2. Restart Mac, it's erroneous

### Expected behaviour

Program icons set to be visible remain visible after restarting Mac or Hidden Bar.

### Environment

**Desktop (please complete the following information):**
 - OS: High Sierra 10.13.6, latest updates

Otherwise program works nicely.

**Comment by pumpkin-zbh** [comment]
Yes, the order of the icons will always be disrupted after the device is restarted, resulting in having to configure again.

**Comment by mcbaer** [comment]
> Yes, the order of the icons will always be disrupted after the device is restarted, resulting in having to configure again.

And yet it doesn't have to work this way.. configuration could be stored, and set upon restart

**Comment by jeannnemelanie** [comment]
I have similar issue but only with 2 apps (Kiwi for Gmail 3.4 & 1Password 8). All the others apps are working well and always keep  the order I defined.

I don't know if I should contact the owners of the 2 apps or if it's a problem related to hidden ?

MacBook Pro 14'' 2021 (M1 Pro Chip)
macOS Monterey 12.2.1

**Comment by viettrungphan** [comment]
System(MacOS) will remember the locations of icons. Some app simple constantly create new icon instead of using the same button. It has same image and functionality (but not same button each time you reopen app). To make system remember location app should assign same id for that button. This should be finished at client app. We can't do anything. That's the reason why you see some app icon not working because new app icon automatically add to left hand side of menu bar.

**Comment by jeannnemelanie** [comment]
Hi @viettrungphan ! Thanks a lot for your clear explanations. I will forward this information to the problematic apps. Again, thank you very much.

**Comment by endore8** [comment]
I'm a developer of [Usage](https://apps.apple.com/app/apple-store/id1561788435?pt=117722237&ct=website&mt=8) app, which has dynamic menu bar items. For example, it shows bluetooth battery of a connected device, and the menu bar item for it is shown only when the device is connected. When the menu bar item appears or disappears (the ids are static), HiddenBar does not remember the position of the any menu bar item, and hides them all.
I've looked in ways to perhaps fix it on my side, but I already set the same menu bar ids when reloading, showing / hiding them. Would be good to hear an opinion of someone who knows HiddenBar's implementation on that.
